File: core/java/src/main/java/hara/lang/protocol/IFlush.java
package hara.lang.protocol;

import hara.lang.declaration.HaraAvailability;
import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

/** Asynchronous buffered-write barrier. */
@HaraProtocolBinding(
    namespace = "std.protocol.iflush",
    name = "IFlush",
    availability = HaraAvailability.CAPABILITY_GATED,
    capability = "native-runtime-protocols")
public interface IFlush {
  @HaraMethod(value = "flush", arity = 1)
  Object flush();
}

File: core/rust/src/lang/protocol/iflush.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(
    namespace = "std.protocol.iflush",
    name = "IFlush",
    availability = "capability-gated",
    capability = "native-runtime-protocols"
)]
pub trait IFlush {
    type Output;

    #[hara_method(value = "flush", arity = 1)]
    fn flush(&mut self) -> Self::Output;
}
